lib/AST/ASTVisitor.cpp

```diff
@@ -0,0 +1,94 @@
+#include "mull/AST/ASTVisitor.h"
+
+#include "mull/Program/Program.h"
+
+#include <clang/AST/RecursiveASTVisitor.h>
+#include <clang/Basic/SourceLocation.h>
+
+using namespace mull;
+
+ASTVisitor::ASTVisitor(mull::Diagnostics &diagnostics, mull::ThreadSafeASTUnit &astUnit,
+                       mull::SingleASTUnitMutations &singleUnitMutations,
+                       mull::FilePathFilter &filePathFilter, mull::MutatorKindSet mutatorKindSet)
+    : diagnostics(diagnostics), astUnit(astUnit), singleUnitMutations(singleUnitMutations),
+      filePathFilter(filePathFilter), sourceManager(astUnit.getSourceManager()),
+      mutatorKindSet(mutatorKindSet), shouldSkipCurrentFunction(false) {}
+
+bool ASTVisitor::VisitFunctionDecl(clang::FunctionDecl *Decl) {
+  if (Decl->getNameAsString() == "main") {
+    shouldSkipCurrentFunction = true;
+  } else {
+    shouldSkipCurrentFunction = false;
+  }
+  return clang::RecursiveASTVisitor<ASTVisitor>::VisitFunctionDecl(Decl);
+}
+
+bool ASTVisitor::VisitExpr(clang::Expr *expr) {
+  if (shouldSkipCurrentFunction) {
+    return true;
+  }
+
+  clang::SourceLocation exprLocation = expr->getBeginLoc();
```

Comment:
LLVM 3.9 compains:

```
 /opt/llvm-3.9.0/bin/clang++  -DSPDLOG_COMPILED_LIB -I/home/runner/work/mull/mull/include -I/home/runner/work/mull/mull/LLVMCompatibility/3.9.x -I/home/runner/work/mull/mull/vendor/spdlog/include -isystem /opt/llvm-3.9.0/include -isystem /home/runner/work/mull/mull/vendor/LibEBC/lib/include -isystem /home/runner/work/mull/mull/vendor/libirm/include -isystem /home/runner/work/mull/mull/vendor -D_GLIBCXX_USE_CXX11_ABI=1 -O3    -std=c++11 -fPIC -fvisibility-inlines-hidden -fno-exceptions -Wunused -Werror -fno-rtti -MD -MT lib/CMakeFiles/mull.dir/AST/ASTVisitor.cpp.o -MF lib/CMakeFiles/mull.dir/AST/ASTVisitor.cpp.o.d -o lib/CMakeFiles/mull.dir/AST/ASTVisitor.cpp.o -c /home/runner/work/mull/mull/lib/AST/ASTVisitor.cpp
    /home/runner/work/mull/mull/lib/AST/ASTVisitor.cpp:31:46: error: no member named 'getBeginLoc' in 'clang::Expr'
      clang::SourceLocation exprLocation = expr->getBeginLoc();
                                           ~~~~  ^
```

Reply:
Btw, if it's only LLVM 3.9 why `LLVM_VERSION_MAJOR >= 7`?

Reply:
I think I simply got confused by the feedback from GitHub Actions and also misinformed you: the LLVM 3.9 simply gets run first so it was the first to be seen failed, but the actual edge is between 6 and 7.